test/ffmpeg-vaapi/encode/avc.py

```diff
@@ -61,6 +61,46 @@ def test_r2r(self, case, gop, slices, bframes, qp, quality, profile):
     vars(self).setdefault("r2r", 5)
     self.encode()
 
+class bdepth(AVCEncoderTest):
+  def init(self, tspec, case, gop, slices, bframes, qp, quality, profile, bdepth):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      bframes   = bframes,
+      case      = case,
+      gop       = gop,
+      profile   = profile,
+      qp        = qp,
+      quality   = quality,
+      rcmode    = "cqp",
+      slices    = slices,
+      bdepth    = bdepth,
+    )
+
+  @slash.parametrize(*gen_avc_bdepth_parameters(spec, ['high', 'main']))
+  def test(self, case, gop, slices, bframes, qp, quality, profile, bdepth):
```

Comment:
How about `(self, case, gop, bframes, bitrate, qp, maxrate, profile, rcmode, bdepth)` instead.  This would allow us to use the same class for different rcmode.  And it would reduce the need for a bunch of specialized generators.
See the `roi` or `intref` test classes for reference.  It would be similar to those (specialized to bdepth cases, of course).